src/adapter/src/coord/appends.rs

```diff
@@ -650,5 +650,6 @@ impl GroupCommitWaiter {
 ///
 /// Note: We sometimes want to throttle how many group commits are running at once, which this
 /// permit allows us to do.
+#[allow(dead_code)]
```

Comment:
Yeah there are a lot of clippy dead_code warnings triggered by the upgrade. I'll try to see if I can figure it out - maybe the clippy version needs updating (if that's possible to do)

Reply:
Ah this is because the field in `GroupCommitPermit` is never accessed/read itself. The same goes for the other structs I had to annotate in this PR (their inner fields are never accessed/read)